enter.c

```diff
@@ -303,7 +303,14 @@ int mutt_enter_string_full(char *buf, size_t buflen, int col, int flags, bool mu
 
         case OP_EDITOR_BACKSPACE:
           if (state->curpos == 0)
-            BEEP();
+          {
+            // Pressing backspace when no text is in the command prompt shold exit the prompt
+            if (state->lastchar == 0)
+              goto bye;
+            // Pressing backspace with text in the command prompt should just beep
+            else
+              BEEP();
```

Comment:
I'd drop the `else` here and unindent the `BEEP`.
In this case it barely matters, but it's good to reduce the indent by returning early.
e.g.
BAD
```c
void function(void)
{
  if (cond)
  {
    // 100s of lines
  }
}
```

GOOD
```c
void function(void)
{
  if (!cond)
    return;

  // 100s of lines
}
```

Reply:
Ok, removed the else as requested.